packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1291,16 +1455,20 @@ class _WidgetsAppState extends State<WidgetsApp> with WidgetsBindingObserver {
 
   @override
   Widget build(BuildContext context) {
-    Widget navigator;
-    if (_navigator != null) {
-      navigator = Navigator(
+    Widget routing;
+    if (_usesRouter) {
+      assert(_effectiveRouteInformationProvider != null);
+      routing = Router<dynamic>(
```

Comment:
i missed that we were creating this with `dynamic` as the type here. That seems to defeat the purpose of having a `T` type argument on Router...

Not sure what the right answer is, but we should probably find a better solution than defining the API with `T` then ignoring it.

(I noticed this because we're changing `dynamic` to `Object` as part of the NNBD transition.)

Reply:
I was struggling hard on this one too, you can not set T for a single constructor, and it is weird to define on the class level because non router constructor does not care. we could separate out router constructor to a class or method on its own, but that seems to add a whole lot of complexity to the code. For now, it will let you initialize with different type of route information parser and router delegate, but it will throw on build